### PR TITLE
SystemReport: Don't call property `parentNode()` and `setAttribute()` on null

### LINKED ISSUE #129: Trying to get property 'parentNode' of non-object when using scheduler for system report
## Describe the bug

We have a system report without a template.

We can manually download the report via the UI.

Running icingacli reporting schedule run results in this error:

`ERROR: ErrorException in /usr/share/icingaweb2/modules/reporting/library/Reporting/Reports/SystemReport.php:27 with message: Trying to get property 'parentNode' of non-object`

Other report types work fine with schedules.

## To Reproduce

1. new report
  2. timeframe: 4 hours
  3. template: empty
  4. report: system
5. create schedule for report
  6. frequency: minutely
  7. action: send mail
  8. type: pdf
9. icingacli schedule run
10. wait

## Expected behavior

Error should not happen

## Your Environment

* Module version: 0.10.0
* Dependent module versions: 
* Icinga Web 2: 2.10.1
* pdfexport: 0.10.1
* Web browser and version: chrome 102.0.5005.63
* Icinga 2 version used (`icinga2 --version`): r2.13.3-1
* PHP version used (`php --version`): 7.3.31-1~deb10u1
* Server operating system and version: Debian 10

### PR DESCRIPTION
**Pull request by jhoxhaa**
fix #129

### CODE REVIEW

**Comment by jhoxhaa** [comment]
@cla-bot

**Comment by jhoxhaa** [comment]
@cla-bot check

**Comment by bobapple** [comment]
@cla-bot check

File: library/Reporting/Reports/SystemReport.php [hunk 1]
```diff
@@ -24,15 +24,24 @@ public function getHtml(Timerange $timerange, array $config = null)
         @$doc->loadHTML($html);
 
         $style = $doc->getElementsByTagName('style')->item(0);
-        $style->parentNode->removeChild($style);
+
```
Comment (lippserd) [inline library/Reporting/Reports/SystemReport.php:27]: Please remove the newline.

File: library/Reporting/Reports/SystemReport.php [hunk 2]
```diff
@@ -24,15 +24,24 @@ public function getHtml(Timerange $timerange, array $config = null)
...
+        if ($style) {
```
Comment (lippserd) [inline library/Reporting/Reports/SystemReport.php:28]: Please use `!== null`.

File: library/Reporting/Reports/SystemReport.php [hunk 3]
```diff
@@ -24,15 +24,24 @@ public function getHtml(Timerange $timerange, array $config = null)
...
+            $style->parentNode->removeChild($style);
+        }
 
         $title = $doc->getElementsByTagName('title')->item(0);
-        $title->parentNode->removeChild($title);
+        if ($title) {
```
Comment (lippserd) [inline library/Reporting/Reports/SystemReport.php:33]: Please use `!== null`.

File: library/Reporting/Reports/SystemReport.php [hunk 4]
```diff
@@ -24,15 +24,24 @@ public function getHtml(Timerange $timerange, array $config = null)
...
+            $title->parentNode->removeChild($title);
+        }
 
         $meta = $doc->getElementsByTagName('meta')->item(0);
-        $meta->parentNode->removeChild($meta);
+        if ($meta) {
```
Comment (lippserd) [inline library/Reporting/Reports/SystemReport.php:38]: Please use `!== null`.

File: library/Reporting/Reports/SystemReport.php [hunk 5]
```diff
@@ -24,15 +24,24 @@ public function getHtml(Timerange $timerange, array $config = null)
...
+            $meta->parentNode->removeChild($meta);
+        }
 
-        $doc->getElementsByTagName('div')->item(0)->setAttribute('class', 'system-report');
+        if (($el = $doc->getElementsByTagName('div')->item(0)) !== null) {
```
Comment (lippserd) [inline library/Reporting/Reports/SystemReport.php:42]: All other blocks also do not set the variable in the `if`, so please do the same here, e.g.

```
$div = ...
if ($div !== null) {
    ...
}

**Review by lippserd — CHANGES_REQUESTED** [review]

**Comment by lippserd** [comment]
@jhoxhaa Can you please check why this is even necessary? According to the issue, this only fails when the report is created from a schedule.

**Review by nilmerg — CHANGES_REQUESTED** [review]
While the error is gone, the resulting report isn't readable anymore now: 
[system (Current Day) 2022-11-04 1038.pdf](https://github.com/Icinga/icingaweb2-module-reporting/files/9936935/system.Current.Day.2022-11-04.1038.pdf)

**Review by nilmerg — APPROVED** [review]

**Review by lippserd — APPROVED** [review]

**Pull request merged**